docs-next/pages/apis/context.mdx

```diff
@@ -2,6 +2,139 @@ import { Markdown } from '../../components/Page';
+`variables` is an optional object containing the input variables for the query.
+The returned object is an [`ExecutionResult.data`](https://graphql.org/graphql-js/execution/).
+Any errors generated by the query will be thrown as an exception.
+This query uses the current `context` object as its `context`.
+
+### Session API
+
+If you configure your Keystone system with session management then you will have access to the following properties.
+See the [session API](./session#session-context) for more details.
+
+`session`: The current session data object.
+
+`startSession`: An internal helper function used by authentication mutations to start a session on a successful login. This should not be called directly.
+
+`endSession`: An internal helper function used by authentication mutations to end a session on logout. This should not be called directly.
+
+### New context creators
+
+When using the `context.lists`, `context.graphql.run`, and `context.graphql.raw` APIs, access control and session information is passed through to these calls from the `context` object.
+The following functions will create a new `KeystoneContext` object with this behaviour modified.
+
+`context.sudo()` returns a new `KeystoneContext` object with all access control disabled for subsequent API calls.
+
+`context.exitSudo()` returns a new `KeystoneContext` object with all access control re-enabled for subsequent API calls.
+
+`context.withSession(newSession)` returns a new `KeystoneContext` object with the `.session` object replaced with `newSession`.
+
+### Database access
+
+The `KeystoneContext` object exposes the underlying database driver directly, depending on the value of [`config.db.adapter`](http://localhost:8000/apis/config#db).
+
+If `config.db.adapter === 'prisma_postgresql'`, then `context.prisma` will be a [Prisma Client](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference) object.
+
+If `config.db.adapter === 'knex'`, then `context.knex` will be a [Knex connection](http://knexjs.org/) object.
+
+if `config.db.adapter === 'mongoose'`, then `context.mongoose` will be a [Mongoose](https://mongoosejs.com/docs/api/mongoose.html) object.
+
+### Access control helpers
+
+The functions `getListAccessControlForUser()` and `getFieldAccessControlForUser()` are used by the internal Keystone resolvers to apply access control.
+They should not be called directly.
+
+### Internal state
+
+These properties are used internally by Keystone and generally do not need to be directly accessed.
+
+`totalResults`: The cumulative total number of results returned by the current request. See [`config.graphql.queryLimits`](./config#graphql).
+
+`maxTotalResults`: The maximum number of results which can be returned before a query limit error is triggered. See [`config.graphql.queryLimits`](./apis/config#graphql).
+
+`schemaName`: The internal name used by Keystone to refer to the GraphQL schema.
```

Comment:
Isn't this also deprecated? (in the sense that yes it's internal state now, but we're going to flatten it out of existence sometime soon)

Reply:
It's complicated. I would like it to be deprecated, it's not *quite* there yet. There a few yaks that need to shave before we can confidently deprecate it. For now, it can stay as `internal state`.

Reply:
I'm... really inclined to _call it deprecated_ even if we're not at the point where we'd actually say we know how we're going to get rid of it. Like, it could be deprecated for a year, that's fine. Maybe one day in a super unexpected outcome we actually realise we should un-deprecate it.

Difference between "we're documenting that you're not meant to use or even really know about this" vs "this is a technical thing that exists"

But I'm also happy for you to make the call